eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,182 @@
+  $yamlContent = $markdownContent -split $splitString
+  foreach($yamlSection in $yamlContent) {
+    if ($yamlSection) {
+      try {
+        # remove the lines like: $(tag) == 'package-preview-2023-09'
+        $yamlSection = $yamlSection -replace '^\s*\$\(.+\)\s*==.+', ''
+        $yamlobj = ConvertFrom-Yaml -Yaml $yamlSection
+        if($yamlobj) {
+          $batchValue = $yamlobj["batch"]
+          $requireValue = $yamlobj["require"]
+          $inputFileValue = $yamlobj["input-file"]
+          if ($requireValue) {
+            LogDebug "require is set as:$requireValue"
+            Verify-AutorestMd $requireValue
+          }
+          elseif ($inputFileValue) {
+            LogDebug "input-file is set as:$inputFileValue"
+            foreach($inputFile in $inputFileValue) {
+              Verify-AutorestMd $inputFile
+            }
+          }
+          elseif ($batchValue) {
+            # there are some services which use batch mode for sdk generation, e.g. Azure.AI.Language.QuestionAnswering
+            foreach($batch in $batchValue) {
+              $requireValue = $batch["require"]
+              $inputFileValue = $batch["input-file"]
+              if ($requireValue) {
+                LogDebug "require is set as:$requireValue"
+                Verify-AutorestMd $requireValue
+              }
+              elseif ($inputFileValue) {
+                LogDebug "input-file is set as:$inputFileValue"
+                foreach($inputFile in $inputFileValue) {
+                  Verify-AutorestMd $inputFile
+                }
+              }
+            }
+          }
+        }
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse yaml section $yamlSection with exception:`n$_ "
+      }
+    }
+  }
+}
+
+try{
+  $ServiceDir = Join-Path $RepoRoot 'sdk' $ServiceDirectory
+  $PackageDirectory = Join-Path $ServiceDir $PackageName
+  Push-Location $PackageDirectory
+
+  # Load tsp-location.yaml if existed
+  $tspLocationYamlPath = Join-Path $PackageDirectory "tsp-location.yaml"
+  $autorestMdPath = Join-Path $PackageDirectory "src/autorest.md"
+  $swaggerReadmePath = Join-Path $PackageDirectory "swagger/README.md"
+  $tspLocationYaml = @{}
+  if (Test-Path -Path $tspLocationYamlPath) {
+    # typespec scenario
+    $tspLocationYaml = Get-Content -Path $tspLocationYamlPath -Raw | ConvertFrom-Yaml
+    Verify-TspLocation $tspLocationYaml
+  }
+  elseif ($Language -eq "dotnet") {
+    # only dotnet language sdk uses 'autorest.md' to configure the sdk generation
+    if (Test-Path -Path $autorestMdPath) {
+      try {
+        $autorestMdContent = Get-Content -Path $autorestMdPath -Raw
+        Verify-Content $autorestMdContent
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse autorest.md file with exception:`n$_ "
+      }
+    }
+  }
+  elseif ($Language -eq "java" -or $Language -eq "js") {
```

Comment:
Is the handling of JS the same with Java?
I found JS is using _meta.json file  [sdk/agrifood/arm-agrifood/_meta.json](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/agrifood/arm-agrifood/_meta.json)

Reply:
Thanks @lirenhe for the reminder. The config format appears to be different from `readme.md` and `autorest.md`.

@qiaozha , can you confirm if `_meta.json` is the configuration file of the spec for JS management plane SDK?

Reply:
we store the commit id for swagger in `_meta.json` , which is for regeneration for codegen change only, if it's for pipeline, I think we should always use the lastest commit id ?

Reply:
@qiaozha 
Do you mean `_meta.json` is only used for re-generation when codegen changes?  
I would like to know is there any config of spec location used in JS SDK release phase? That would be the content of this validation to check.  
Or, does the JS generation script pull the spec from latest main branch of Azure/azure-rest-api-specs repo by default for release? If that's the case, then we can ignore this validation for JS mgmt. plane scenario.

Reply:
@lirenhe , for Java/JS/Go mgmt. plane scenarios, they will always get the latest main branch of Azure/azure-rest-api-specs repo to release the SDK. I will ignore the validation for them.